lib/opal/cli_runners/firefox.rb

```diff
@@ -134,6 +134,15 @@ def run_firefox_server
       rescue Timeout::Error
         puts 'Failed to start firefox server'
         puts 'Make sure that you have it installed and that its version is > 100'
+        if !OS.windows? && !OS.macos?
+          # The firefox executable within snap is in fact the snap executable which sets up paths and then calls the
+          # real firefox executable. It also mistreats passed options and args and always tries to reuse a existing
+          # instance. Thus firefox from snap, when called by this runner, will start with the wrong options, the
+          # wrong profile and will fail to setup the remote debugging port
```

Comment:
Wild idea: What if we resolve a correct path to Firefox from snap, ie. /snap/asdad/bin/firefox and call it directly?

Reply:
I had that wild idea. The question is how and how to make that reliable?